includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -137,21 +173,26 @@ public function capture_terminal_payment( $request ) {
 			}
 
 			// Update the order: set the payment method and attach intent attributes.
-			$order->set_payment_method( WC_Payment_Gateway_WCPay::GATEWAY_ID );
-			$order->set_payment_method_title( __( 'WooCommerce In-Person Payments', 'woocommerce-payments' ) );
+			if ( $is_terminal_payment ) {
+				$order->set_payment_method( WC_Payment_Gateway_WCPay::GATEWAY_ID );
+				$order->set_payment_method_title( __( 'WooCommerce In-Person Payments', 'woocommerce-payments' ) );
+			}
 			$intent_id     = $intent->get_id();
```

Comment:
We can move the declaration for these four variables above and merge the below condition with the one above.

Reply:
Good point thanks!